.ci/generate-extra-site-links.sh

```diff
@@ -19,11 +19,12 @@ echo "PR_NUMBER=$PR_NUMBER"
 echo "AWS_FOLDER_LINK=$AWS_FOLDER_LINK"
 
 # Extract a list of the changed xdocs in the pull request. For example 'src/xdocs/config_misc.xml'.
+# We ignore template files and deleted files.
 CHANGED_XDOCS_PATHS=$(curl --fail-with-body -Ls \
   -H "Accept: application/vnd.github+json" \
   -H "Authorization: Bearer $GITHUB_TOKEN" \
   "https://api.github.com/repos/$REPOSITORY_OWNER/checkstyle/pulls/$PR_NUMBER/files?per_page=100" |
-  jq -r ".[] | .filename" | grep src/xdocs/ || true)
+  jq -r ".[] | .filename" | grep src/xdocs/ | grep -v '.*xml.template$\|.*#$' || true)
```

Comment:
Essentially we are removed any results that are a) template files, or b) end with `#`, which has been shown to indicate that a file has been deleted.

Reply:
Ah, ok, here is one of the action generated comments: https://github.com/checkstyle/checkstyle/pull/13103#issuecomment-1572524802

Not all links ending with `#` are broken.

Reply:
I believe this `#` comes from
https://github.com/checkstyle/checkstyle/blob/ba9e673d020ceb04ba66e2fa2f6e9f9300113b44/.ci/generate-extra-site-links.sh#L67-L68

There is no `$SUBSECTION_ID` so the `#` is left alone. I think there was a comment about it a while ago but it was alright to leave it since the URL still works with a `#` at the end. I guess it created confusion here.

Reply:
Yeah, we need to gather this info from the diff then. I am on it.

Reply:
Proof that update works, used @stoyanK7 's example from above:
```bash
➜  checkstyle git:(issue-13175-testing) ✗ cat test.json | jq -r '.[] | select(.status != "removed") | .filename' 

src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```

We filter the jq stream by removing all list elements with status as "removed".